### PR TITLE
Alternative sniper rifle

### PR DESCRIPTION
**Pull request by MsSeraph**
Basically i added an alternative syndie sniper rifle which nuke ops can buy, mostly a reskin.
pros: it holds 4 more rounds, and costs 2 tc less
cons: it uses bolt action, does less damage and does not dismember
neutral: looks cool and uses counter strike 1.6 sounds.
PS: the "lathe gun" stuff is because i was on the lathe gun branch when i made this one, however i removed all code regarding the lathe gun (yes even the sprites).
#### Changelog

:cl:  
rscadd: Added the AWP, an alternative to the syndicate sniper rifle that uses Counter Strike 1.6 sound effects.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 1]
```diff
@@ -344,3 +344,71 @@
...
+	fire_delay = 40
+	burst_size = 1
+	w_class = WEIGHT_CLASS_NORMAL
+	zoomable = TRUE
+	zoom_amt = 10
+	zoom_out_amt = 13
+	slot_flags = ITEM_SLOT_BACK
+	actions_types = list()
+	mag_display = TRUE
+	bolt_type = BOLT_TYPE_STANDARD
```
Comment (Hopekz) [inline code/modules/projectiles/guns/ballistic/automatic.dm]:
```suggestion
```
Redundant.
You already define this as BOLT_TYPE_STANDARD from your ballistic parent.

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 2]
```diff
@@ -344,3 +344,71 @@
...
+// AWP //
+
+/obj/item/gun/ballistic/automatic/sniper_rifle/alt
+	name = "AWP"
+	desc = "A powerful sniper rifle chambered in .338 lapua magnum. It kinda smells like tears of silvers, whatever that means."
+	icon_state = "sniper_alt"
+	item_state = "sniper_alt"
+	fire_sound = 'sound/weapons/awp_fire.ogg'
+	fire_sound_volume = 100
+	vary_fire_sound = FALSE
```
Comment (Hopekz) [inline code/modules/projectiles/guns/ballistic/automatic.dm]:
```suggestion
```
Redundant.

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 3]
```diff
@@ -344,3 +344,71 @@
...
+	load_sound = 'sound/weapons/awp_clipin.ogg'
+	rack_sound = 'sound/weapons/boltpull.ogg'
+	recoil = 2
```
Comment (Hopekz) [inline code/modules/projectiles/guns/ballistic/automatic.dm]:
```suggestion
```
Redundant.

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 4]
```diff
@@ -344,3 +344,71 @@
...
+	weapon_weight = WEAPON_HEAVY
```
Comment (Hopekz) [inline code/modules/projectiles/guns/ballistic/automatic.dm]:
```suggestion
```
Redundant.

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 5]
```diff
@@ -344,3 +344,71 @@
...
+	mag_type = /obj/item/ammo_box/magazine/sniper_rounds/alt
+	fire_delay = 40
+	burst_size = 1
+	w_class = WEIGHT_CLASS_NORMAL
+	zoomable = TRUE
```
Comment (Hopekz) [inline code/modules/projectiles/guns/ballistic/automatic.dm]:
```suggestion
```
Redundant.

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 6]
```diff
@@ -344,3 +344,71 @@
...
+	item_state = "sniper_alt"
+	fire_sound = 'sound/weapons/awp_fire.ogg'
+	fire_sound_volume = 100
+	vary_fire_sound = FALSE
+	load_sound = 'sound/weapons/awp_clipin.ogg'
+	rack_sound = 'sound/weapons/boltpull.ogg'
+	recoil = 2
+	weapon_weight = WEAPON_HEAVY
+	mag_type = /obj/item/ammo_box/magazine/sniper_rounds/alt
+	fire_delay = 40
```
Comment (Hopekz) [inline code/modules/projectiles/guns/ballistic/automatic.dm]:
```suggestion
```
Redundant.

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 7]
```diff
@@ -344,3 +344,71 @@
...
+	burst_size = 1
```
Comment (Hopekz) [inline code/modules/projectiles/guns/ballistic/automatic.dm]:
```suggestion
```
Redundant.

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 8]
```diff
@@ -344,3 +344,71 @@
...
+	w_class = WEIGHT_CLASS_NORMAL
+	zoomable = TRUE
+	zoom_amt = 10
```
Comment (Hopekz) [inline code/modules/projectiles/guns/ballistic/automatic.dm]:
```suggestion
```
Redundant.

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 9]
```diff
@@ -344,3 +344,71 @@
...
+	zoom_out_amt = 13
+	slot_flags = ITEM_SLOT_BACK
```
Comment (Hopekz) [inline code/modules/projectiles/guns/ballistic/automatic.dm]:
```suggestion
```
Redundant.

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 10]
```diff
@@ -344,3 +344,71 @@
...
+	actions_types = list()
+	mag_display = TRUE
```
Comment (Hopekz) [inline code/modules/projectiles/guns/ballistic/automatic.dm]:
```suggestion
```
Redundant.

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 11]
```diff
@@ -344,3 +344,71 @@
...
+	rack_sound = 'sound/weapons/boltpull.ogg'
+	recoil = 2
+	weapon_weight = WEAPON_HEAVY
+	mag_type = /obj/item/ammo_box/magazine/sniper_rounds/alt
+	fire_delay = 40
+	burst_size = 1
+	w_class = WEIGHT_CLASS_NORMAL
+	zoomable = TRUE
+	zoom_amt = 10
+	zoom_out_amt = 13
```
Comment (Hopekz) [inline code/modules/projectiles/guns/ballistic/automatic.dm]:
```suggestion
```
Redundant.

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 12]
```diff
@@ -344,3 +344,71 @@
...
+	fire_sound_volume = 100
+	vary_fire_sound = FALSE
+	load_sound = 'sound/weapons/awp_clipin.ogg'
+	rack_sound = 'sound/weapons/boltpull.ogg'
+	recoil = 2
+	weapon_weight = WEAPON_HEAVY
+	mag_type = /obj/item/ammo_box/magazine/sniper_rounds/alt
+	fire_delay = 40
+	burst_size = 1
+	w_class = WEIGHT_CLASS_NORMAL
```
Comment (Hopekz) [inline code/modules/projectiles/guns/ballistic/automatic.dm]:
```suggestion
```
Redundant from parent.

**Review by Hopekz — COMMENTED** [review]
You have a lot of redundant code.
All of this is already defined in your parents ( sniper_rifle and ballistic)

**Comment by Hopekz** [comment]
You took code from bolt action rifles just so you can put this rifle in automatic?
It makes no sense just move it to bolt action.

Now if someone modifies bolt action rifles your rifle is a special snowflake and won't be touched and there seems to be no reason for it except "The other sniper rifle is here so I put it here"

**Comment by Ktlwjec1** [comment]
code/modules/projectiles/guns/ballistic/rifle.dm:193:error: tac_load: undefined var
code/modules/projectiles/guns/ballistic/rifle.dm:194:error: user: undefined var
code/modules/projectiles/guns/ballistic/rifle.dm:194:error: tac_load: undefined var
code/modules/projectiles/guns/ballistic/rifle.dm:195:error: user: undefined var
code/modules/projectiles/guns/ballistic/rifle.dm:197:error: user: undefined var
code/modules/projectiles/guns/ballistic/rifle.dm:201:error: user.put_in_hands: undefined var
code/modules/projectiles/guns/ballistic/rifle.dm:203:error: display_message: undefined var
code/modules/projectiles/guns/ballistic/rifle.dm:204:error: user: undefined var
code/modules/projectiles/guns/ballistic/rifle.dm:184:error: eject_magazine: duplicate definition
code/modules/projectiles/guns/ballistic.dm:176:error: eject_magazine: previous definition

**Comment by MsSeraph** [comment]
Its got little to no redundant Code, its in Bolt action, It works, what else do you want from me, i literally started Coding a week ago

**Comment by MsSeraph** [comment]
Finished at last. Also Hopek you managed to break my PR twice lol xd

File: code/modules/projectiles/guns/ballistic/rifle.dm [hunk 13]
```diff
@@ -143,3 +143,66 @@ obj/item/gun/ballistic/rifle/attackby(obj/item/A, mob/user, params)
...
+/obj/item/gun/ballistic/automatic/sniper_rifle/alt/equipped(mob/user, slot)
+	if(!ishuman(user))
+		return
+	if(slot == SLOT_HANDS)
+		playsound(loc, 'sound/weapons/awp_deploy.ogg', 100, FALSE, 8)
+	return
+
+obj/item/gun/ballistic/automatic/sniper_rifle/alt/rack(mob/user = null)
+	if (bolt_locked == FALSE)
+		to_chat(user, "<span class='notice'>You open the bolt of \the [src]</span>")
```
Comment (Ktlwjec1) [inline code/modules/projectiles/guns/ballistic/rifle.dm:183]: full stop -> [src].

File: code/modules/uplink/uplink_items.dm [hunk 14]
```diff
@@ -493,6 +493,16 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 	surplus = 25
 	include_modes = list(/datum/game_mode/nuclear)
 
+/datum/uplink_item/dangerous/sniper/alt
+	name = "Alternative Sniper Rifle"
+	desc = "Who needs a modern .50 sniper rifle when you can get a classic .338 lapua magnum sniper rifle!.\
```
Comment (Ktlwjec1) [inline code/modules/uplink/uplink_items.dm:498]: rifle? or rifle. -> has both !. currently, "!" doesn't make sense.

**Review by Ktlwjec1 — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/guns/ballistic/rifle.dm [hunk 15]
```diff
@@ -143,3 +143,66 @@ obj/item/gun/ballistic/rifle/attackby(obj/item/A, mob/user, params)
...
 	else
 		user.dropItemToGround(src, TRUE)
+
+///////
+//AWP//
+///////
+
+/obj/item/gun/ballistic/automatic/sniper_rifle/alt
+	name = "AWP"
+	desc = "A powerful sniper rifle chambered in .338 lapua magnum. It kinda smells like tears of silvers, whatever that means."
```
Comment (Ktlwjec1) [inline code/modules/projectiles/guns/ballistic/rifle.dm:153]: Also I don't know if this is a reference to something but it doesn't make sense grammatically anyway, needs to be "silver" not "silvers".

**Review by Ktlwjec1 — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/guns/ballistic/rifle.dm (hunk 15 above)
Comment (MsSeraph) [reply code/modules/projectiles/guns/ballistic/rifle.dm:153]: Silvers as in the silver rank of CS GO, not everything i do is wrong.

Comment (swissloaf) [reply code/modules/projectiles/guns/ballistic/rifle.dm:153]: silvers are a group of people @Ktlwjec1 know ur stuff

**Review by monster860 — CHANGES_REQUESTED** [review]
delete awp.dmi, it's not used

**Comment by Hopekz** [comment]
You added your coilgun_fire.ogg sound but it is not being used anywhere in your PR.

**Comment by Ktlwjec1** [comment]
Going to close this due to no updates in 12 days (stale label should last around a week and it's been almost 2) - travis no likely since dme files get deleted for no reason and I haven't checked but what hopek said will need to be addressed too if that's the case. Feel free to reopen/make a new PR when you have time to address these issues.

**Pull request closed without merging**